File: _includes/code/java-v6/src/test/java/ConfigureBQTest.java
import io.weaviate.client6.v1.api.WeaviateClient;
import io.weaviate.client6.v1.api.collections.CollectionHandle;
import io.weaviate.client6.v1.api.collections.Property;
import io.weaviate.client6.v1.api.collections.Quantization;
import io.weaviate.client6.v1.api.collections.VectorConfig;
import io.weaviate.client6.v1.api.collections.vectorindex.Hnsw;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.util.Map;

class ConfigureBQTest {

  private static WeaviateClient client;

  @BeforeAll
  public static void beforeAll() {
    // START ConnectCode
    String openaiApiKey = System.getenv("OPENAI_API_KEY");
    client = WeaviateClient
        .connectToLocal(config -> config.setHeaders(Map.of("X-OpenAI-Api-Key", openaiApiKey)));
    // END ConnectCode
  }

  @AfterAll
  public static void afterAll() throws Exception {
    client.close();
  }

  @Test
  void testEnableBQ() throws IOException {
    String collectionName = "MyCollection";
    if (client.collections.exists(collectionName)) {
      client.collections.delete(collectionName);
    }

    // START EnableBQ
    client.collections.create("MyCollection",
        col -> col.vectorConfig(VectorConfig.text2vecContextionary(vc -> vc
            // highlight-start
            .quantization(Quantization.bq())
        // highlight-end
        )).properties(Property.text("title")));
    // END EnableBQ
  }

  // TODO[g-despot] Errors on collection update: TYPE_UPDATE_CLASS: bad request :parse class update: invalid update for vector "default": skipDefaultQuantization is immutable: attempted change from "true" to "false"
  @Test
  void testUpdateSchema() throws IOException {
    String collectionName = "MyCollection";
    if (client.collections.exists(collectionName)) {
      client.collections.delete(collectionName);
    }
    client.collections.create(collectionName,
        col -> col
            .vectorConfig(VectorConfig
                .text2vecContextionary(vc -> vc.quantization(Quantization.uncompressed())))
            .properties(Property.text("title")));

    // START UpdateSchema
    CollectionHandle<Map<String, Object>> collection = client.collections.use("MyCollection");
    collection.config.update(collectionName, c -> c.vectorConfig(
        VectorConfig.text2vecContextionary(vc -> vc.quantization(Quantization.bq()))));
    // END UpdateSchema
    // TODO[g-despot]: Verify the update
  }

  @Test
  void testBQWithOptions() throws IOException {
    String collectionName = "MyCollection";
    if (client.collections.exists(collectionName)) {
      client.collections.delete(collectionName);
    }

    // START BQWithOptions
    client.collections.create("MyCollection",
        col -> col.vectorConfig(VectorConfig.text2vecContextionary(vc -> vc
            // highlight-start
            .quantization(Quantization.bq(q -> q.cache(true).rescoreLimit(200)))
            .vectorIndex(Hnsw.of(c -> c.vectorCacheMaxObjects(100000)))
        // highlight-end
        )).properties(Property.text("title")));
    // END BQWithOptions
  }
}

File: _includes/code/csharp/ConfigureBQTest.cs
using Xunit;
using Weaviate.Client;
using Weaviate.Client.Models;
using System;
using System.Threading.Tasks;

namespace WeaviateProject.Tests;

public class ConfigureBQTest : IAsyncLifetime
{
    private WeaviateClient client;
    private const string COLLECTION_NAME = "MyCollection";

    // Runs before each test
    public async Task InitializeAsync()
    {
        // START ConnectCode
        client = await Connect.Local();
        // END ConnectCode
    }

    // Runs after each test
    public Task DisposeAsync()
    {
        // No action needed here, as cleanup happens in InitializeAsync before the next test.
        return Task.CompletedTask;
    }

    [Fact]
    public async Task TestEnableBQ()
    {
        await client.Collections.Delete(COLLECTION_NAME);
        // START EnableBQ
        await client.Collections.Create(new CollectionConfig
        {
            Name = "MyCollection",
            Properties = [Property.Text("title")],
            VectorConfig = Configure.Vectors.Text2VecTransformers().New(
                "default",
                new VectorIndex.HNSW
                {
                    // highlight-start
                    Quantizer = new VectorIndex.Quantizers.BQ()
                    // highlight-end
                }
            )
        });
        // END EnableBQ
    }

    [Fact]
    public async Task TestUpdateSchema()
    {
        await client.Collections.Delete(COLLECTION_NAME);
        // Note: Updating quantization settings on an existing collection is not supported by Weaviate
        // and will result in an error, as noted in the Java test. This test demonstrates the syntax for attempting the update.
        var collection = await client.Collections.Create(new CollectionConfig
        {
            Name = "MyCollection",
            Properties = [Property.Text("title")],
            VectorConfig = Configure.Vectors.Text2VecTransformers().New("default")
        });

        // START UpdateSchema
        await collection.Config.Update(c =>
        {
            var vectorConfig = c.VectorConfig["default"];
            vectorConfig.VectorIndexConfig.UpdateHNSW(h => h.Quantizer = new VectorIndex.Quantizers.BQ());
        });
        // END UpdateSchema
    }

    [Fact]
    public async Task TestBQWithOptions()
    {
        await client.Collections.Delete(COLLECTION_NAME);
        // START BQWithOptions
        await client.Collections.Create(new CollectionConfig
        {
            Name = "MyCollection",
            Properties = [Property.Text("title")],
            VectorConfig = Configure.Vectors.Text2VecTransformers().New(
                "default",
                // highlight-start
                new VectorIndex.HNSW
                {
                    VectorCacheMaxObjects = 100000,
                    Quantizer = new VectorIndex.Quantizers.BQ
                    {
                        Cache = true,
                        RescoreLimit = 200
                    }
                }
                // highlight-end
            )
        });
        // END BQWithOptions
    }
}